processor/transformprocessor/internal/metrics/processor.go

```diff
@@ -0,0 +1,107 @@
+// Copyright  The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package metrics // import "github.com/open-telemetry/opentelemetry-collector-contrib/processor/transformprocessor/internal/metrics"
+
+import (
+	"context"
+
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/pdata/pmetric"
+	"go.uber.org/zap"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/processor/transformprocessor/internal/common"
+)
+
+type Processor struct {
+	queries []common.Query
+	logger  *zap.Logger
+}
+
+func NewProcessor(statements []string, functions map[string]interface{}, settings component.ProcessorCreateSettings) (*Processor, error) {
+	queries, err := common.ParseQueries(statements, functions, ParsePath)
+	if err != nil {
+		return nil, err
+	}
+	return &Processor{
+		queries: queries,
+		logger:  settings.Logger,
+	}, nil
+}
+
+func (p *Processor) ProcessMetrics(_ context.Context, td pmetric.Metrics) (pmetric.Metrics, error) {
+	ctx := metricTransformContext{}
+	for i := 0; i < td.ResourceMetrics().Len(); i++ {
+		rmetrics := td.ResourceMetrics().At(i)
+		ctx.resource = rmetrics.Resource()
+		for j := 0; j < rmetrics.ScopeMetrics().Len(); j++ {
+			il := rmetrics.ScopeMetrics().At(j).Scope()
+			ctx.il = il
+			metrics := rmetrics.ScopeMetrics().At(j).Metrics()
+			for k := 0; k < metrics.Len(); k++ {
+				metric := metrics.At(k)
+				ctx.metric = metric
+				switch ctx.metric.DataType() {
```

Comment:
Inconsistent, the switch is using ctx.metric but all of the cases are using metric.Type().DataPoints. My opinion would be just assign directly `ctx.metric = metrics.At(k)` and only use ctx.metric, but I think using metric for the switch would be fine too, I just think it should be consistent within the scope of the switch statement for clarity.

Reply:
Yes i will clean up the assignments/uses

Reply:
Cleaned up